healpy/newvisufunc.py

```diff
@@ -150,45 +237,82 @@ def mollview(
+                ax.set_longitude_grid(longitude_grid_spacing)
+                ax.set_latitude_grid(latitude_grid_spacing)
                 ax.set_longitude_grid_ends(90)
-
-        if graticule_labels:
-            ax.xaxis.set_major_formatter(ThetaFormatterShiftPi(longitude_grid_spacing))
-        else:
+         #       if width < 10:
+          #          ax.set_latitude_grid(45)
+           #         ax.set_longitude_grid_ends(90)
+            else:
+                longitude_grid_spacing = longitude_grid_spacing  # deg
+                latitude_grid_spacing = latitude_grid_spacing #  deg
+                ax.xaxis.set_major_locator(MultipleLocator(np.deg2rad(longitude_grid_spacing) )) # longitude
+                ax.yaxis.set_major_locator(MultipleLocator(np.deg2rad(latitude_grid_spacing))) # lattitude
+
+        if graticule_labels & graticule:
+            if projection_type == "aitoff" or projection_type ==  "hammer" or  projection_type == "lambert" or projection_type == "mollweide":
+                ax.xaxis.set_major_formatter(ThetaFormatterShiftPi(longitude_grid_spacing))
+                ax.tick_params(axis='x', colors=lcolor)
+            else:
+                ax.xaxis.set_major_formatter(ThetaFormatterShiftPi(longitude_grid_spacing))
+                ax.yaxis.set_major_formatter(ThetaFormatterTheta(latitude_grid_spacing))
+        if graticule == False:
             # remove longitude and latitude labels
             ax.xaxis.set_ticklabels([])
             ax.yaxis.set_ticklabels([])
-
+            ax.tick_params(axis=u'both', which=u'both',length=0)
+        
+        ax.set_title(title, fontsize= 14)
         # colorbar
-        if cbar and not m is None:
-            cb = fig.colorbar(
-                ret, orientation="horizontal", shrink=0.4, pad=0.05, ticks=[min, max]
-            )
+        if projection_type is "cart":
+            ax.set_aspect(1)
+        extend = 'neither'
+        if min > np.min(m):
+            extend = "min"
+        if max < np.max(m):
+            extend = "max"
+        if min > np.min(m) and max < np.max(m):
+            extend = 'both'
+        cb = fig.colorbar(
+            ret, orientation=cb_orientation, shrink=shrink, pad=pad, ticks=[min, max], extend=extend
+        )
+        if cb_orientation == 'horizontal':
             cb.ax.xaxis.set_label_text(unit)
-            cb.ax.xaxis.labelpad = -8
-            # workaround for issue with viewers, see colorbar docstring
-            cb.solids.set_edgecolor("face")
-
+            cb.ax.xaxis.labelpad = lpad
+        if cb_orientation == 'vertical':
+            cb.ax.yaxis.set_label_text(unit)
+            cb.ax.yaxis.labelpad = lpad
+        # workaround for issue with viewers, see colorbar docstring
+        cb.solids.set_edgecolor("face")
+        ax.set_xlabel(xlabel,fontsize=12)
+        ax.set_ylabel(ylabel,fontsize=12)
         plt.draw()
     finally:
-        ax.hold(washold)
+        #ax.hold(washold)
+        None
```

Comment:
remove the finally clause

Reply:
`try` has to have an end clause, so I put there `except: pass` instead the finally

Reply:
is this `try-except` even needed? I commented it out in the latest commit.